### ISSUE TITLE
maybe a bug in pandas.Timestamp.strptime

### ISSUE THREAD

**Issue by martinLew** [pandas-dev/pandas#14646]
#### A small, complete example of the issue

```python
# Your code here
pd.Timestamp.strptime('20150303', '%Y%m%d')

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/tslib.pyx", line 294, in pandas.tslib.Timestamp.__new__ (pandas/tslib.c:8967)
TypeError: __new__() takes at most 5 positional arguments (8 given)
```

#### Expected Output
Timestamp('2015-03-03 00:00:00')
#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.11.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-47-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.0
nose: 1.3.7
pip: 9.0.1
setuptools: 27.3.0
Cython: 0.23.1
numpy: 1.11.1
scipy: 0.17.0
statsmodels: 0.6.1
xarray: None
IPython: 4.1.2
sphinx: 1.4.6
patsy: 0.4.1
dateutil: 2.5.1
pytz: 2016.2
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.5
matplotlib: 1.5.1
openpyxl: 2.3.2
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.8.4
lxml: 3.6.0
bs4: 4.4.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.12
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.39.0

I can't trace the code in tslib.c. the result in _strptime.py(462) was still right : (Pdb) year
2015
(Pdb) month
1
(Pdb) day
2

but after the "return" finish, the above Exception occur
</details>

**Comment by chris-b1** [comment]
This seems to be fixed in `0.19` / master - can you try upgrading?

```
In [1]: pd.Timestamp.strptime('20150303', '%Y%m%d')
Out[1]: Timestamp('2015-03-03 00:00:00')

In [2]: pd.__version__
Out[2]: u'0.19.0'
```

**Comment by martinLew** [comment]
I upgraded to 0.19 and that works, thanks a lot
